poetry/semver/__init__.py

```diff
@@ -161,4 +161,7 @@ def parse_single_constraint(constraint):  # type: (str) -> VersionConstraint
         else:
             return version
 
+    if constraint == "<empty>":
+        return VersionRange()
```

Comment:
An empty constraint is not the same thing as a wildcard (`*`) constraint. An empty constraint is an unsolvable constraint which always evaluates to false. This can happen when intersecting to incompatible constraints.

This is not the proper way to fix this issue

Reply:
Fair enough, but then I'm not sure how to really fix the problem then.

Reply:
A comment by @bjoernpollex https://github.com/sdispater/poetry/issues/804#issuecomment-512263672 asks if having `<empty>` be inserted is unintentional.  If `<empty>` being in the lock file is indeed unintentional, it would follow that `<empty>` does not need to be handled later (as it would not be there).

Reply:
The `<empty>` is not even in the lock file. It's being generated by poetry internally, and then when it tries to parse it again, it fails. I'll try to come up with a minimal reproduction. In my case it happens, as @sdispater described, when intersecting different constraints. However, the constraints being intersected are `extras`, and I'm not sure what the proper semantics for those are.